Plugins: added Resource CircularChart example plugin

Faakhir30 opened the pull request:
Closes #586 **[READY FOR REVIEW]**

Implements a Resource CircularChart example plugin showing pod failure stats

This PR adds a new example plugin that demonstrates how to use Headlamp's CircularChart components to visualize cluster resource data. The implementation process involved:

1. Created a new resource-charts plugin showing failed vs total pods
2. Initially used `CircularChart` component, then simplified to use `TileChart` directly:
   - Removed unnecessary metrics-related functionality
   - Added theme-based error coloring for better visualization
   - Improved chart labeling and percentage calculations

3. Added documentation in README.md explaining:
   - Plugin purpose and features
   - Setup instructions
   - Implementation details and key concepts

The example plugin is:
- Using `K8s.ResourceClasses` for data fetching
- Implementing chart `processors` for `Overview ` page as an example page.
- Using `redux` slice for plugin charts, then processing on frontend code.

#### Pod Failed Chart on Overview Page:
![image](https://github.com/user-attachments/assets/a15ddc25-df62-4dda-9bb1-af997a7eab81)

Plugin Details:
![image](https://github.com/user-attachments/assets/886b6aec-e2df-4511-a1e6-8b3f073f5249)

#### Screencast of failing live Pod:
[Screencast from 2025-02-03 02-45-13.webm](https://github.com/user-attachments/assets/0b35d444-f1e0-4f71-a9a5-5815c803ab17)

#### Tests Passing (YES :) 
![image](https://github.com/user-attachments/assets/37107596-b27e-4111-9dc5-60d3d64ba189)

illume commented:
Nice one.

I noticed there's an issue when building docs on CI: https://github.com/headlamp-k8s/headlamp/actions/runs/13102964037/job/36701030284?pr=2832#step:4:28

Probably you can `make docs` to reproduce it locally.

illume commented:
btw, we don't use the `feat()` style commit messages.

Faakhir30 commented:
Thanks @illume ,
Reproduced and fixed the failing check.

Rebased with upstream `main` and cleared commit history to single commit with specified style of commits in headlamp.

illume commented on plugins/examples/resource-charts/src/index.tsx:
```diff
@@ -0,0 +1,52 @@
+import { Box, Paper } from '@mui/material';
+import { useTheme } from '@mui/material/styles';
+import React from 'react';
+
```
I think a short comment here explaining what this plugin is showing would be good. Something like: 

> This shows how to add a custom overview chart on the overview page.

illume commented on plugins/examples/resource-charts/src/index.tsx:
```diff
@@ -0,0 +1,52 @@
+import { useTheme } from '@mui/material/styles';
+import React from 'react';
+
+function PodFailureChart() {
```
A short component tsdoc string would help here.

> PodFailureChart shows the percentage of failing pods.

illume reviewed (approved):
Thanks! :tada:

I tested it, and it works nicely.

ps. I left a couple of optional suggestions for small comments in the index.tsx file which might be a small improvement.

illume commented:
![image](https://github.com/user-attachments/assets/7fa64bf4-250b-4768-9d45-6d1074c0638d)

illume commented:
@joaquimrocha I requested a review from you since it was you who wrote the original issue.
- https://github.com/headlamp-k8s/headlamp/issues/586

Faakhir30 commented:
> ps. I left a couple of optional suggestions for small comments in the index.tsx file which might be a small improvement.

Thanks for the review, added comments/tsdoc string in last separate single commit for easy review (if needed).

illume commented:
> Thanks for the review, added comments/tsdoc string in last separate single commit for easy review (if needed).

Awesome, thanks. Those look good to me.

A small nit: In the linux style git commit messages there is no "." at the end of the title line.

yolossn reviewed (approved):
Great work @Faakhir30, thanks for contributing.

joaquimrocha reviewed (dismissed):
Nice feature @Faakhir30 ! The only thing I think is needed is that the second commit should be actually squashed with the 1st and then the actual functionality of adding a way to process the charts should be split from the example plugin. Because right now the commit says it's about the example plugin but it's actually also adding the functionality.
So I think the 2 final commits should be named something like:
`frontend: Add a way for plugins to process the cluster overview's charts`
`frontend plugins/examples: Add a resource charts example plugin`

Faakhir30 commented:
> So I think the 2 final commits should be named something like: `frontend: Add a way for plugins to process the cluster overview's charts` `frontend plugins/examples: Add a resource charts example plugin`

Updated commits to be atomic as described, thanks.
@illume , finally, PR is ready to merge now ig.
